src/app/ngrx/reducers/index.ts

```diff
@@ -88,5 +96,51 @@ export const selectPodcastMetricsFilteredTotal = createSelector(selectPodcastMet
       return metricsUtil.getTotal(data);
     }
   });
+export const selectPodcastMetricsLoading = createSelector(selectPodcastMetrics, (metrics: PodcastMetricsModel[]) => {
+  return metrics.some((m: PodcastMetricsModel) => m.loading);
+});
+export const selectPodcastMetricsLoaded = createSelector(selectPodcastMetrics, (metrics: PodcastMetricsModel[]) => {
+  return metrics.every((m: PodcastMetricsModel) => m.loaded || m.loaded === undefined);
+});
+const errorType = (code) => code === 401 ? 'Authorization' : 'Unknown';
+export const selectPodcastMetricsError = createSelector(selectPodcastMetrics, (metrics: PodcastMetricsModel[]) => {
+  return metrics.filter(m => m.error).map(m => {
+    return `${errorType(m.error.status)} error occurred while requesting podcast metrics`;
+  });
+});
 
 export const selectEpisodeMetrics = createFeatureSelector<EpisodeMetricsModel[]>('episodeMetrics');
+export const selectEpisodeMetricsLoading = createSelector(selectEpisodeMetrics, (metrics: EpisodeMetricsModel[]) => {
+  return metrics.some((m: EpisodeMetricsModel) => m.loading);
+});
+export const selectEpisodeMetricsLoaded = createSelector(selectEpisodeMetrics, (metrics: EpisodeMetricsModel[]) => {
+  return metrics.every((m: EpisodeMetricsModel) => m.loaded || m.loaded === undefined);
+});
+export const selectEpisodeMetricsError = createSelector(selectEpisodeMetrics, (metrics: EpisodeMetricsModel[]) => {
+  return metrics.filter(m => m.error).map(m => {
+    return `${errorType(m.error.status)} error occurred while requesting episode metrics for ${m.guid}`;
+  });
+});
+
+export const selectCmsLoading = createSelector(selectEpisodesLoading, selectPodcastsLoading, (episodes, podcasts) => episodes || podcasts);
+export const selectCastleLoading = createSelector(selectEpisodeMetricsLoading, selectPodcastMetricsLoading,
```

Comment:
Ahhh... yes.  And this aggregates podcast + all-episodes loading state.  All the things!

Reply:
yep, needed to compose it down to a single observable for `| async` on the spinner